docs/OnClusterBuildInstructions.md

```diff
@@ -0,0 +1,369 @@
+   can verify this by running the following:
+
+   ```console
+   $ oc debug node/node-name
+   Starting pod/node-name-debug ...
+   To use host binaries, run `chroot /host`
+   Pod IP: 10.0.29.42
+   If you don't see a command prompt, try pressing enter.
+   sh-4.4# chroot /host
+   sh-5.1# rpm-ostree status
+   State: idle
+   Deployments:
+   * ostree-unverified-registry:quay.io/myorg/myrepo@sha256:324f17d18b197a951e7e11ea0b101836312191f92aefa0fc2ee240354cfbf7
+                   Digest: sha256:324f17d18b197a951e7e11ea0b101836312191f92aefa0fc2ee240354cfbf7
+                   Version: 414.92.202308210204-0 (2023-08-23T12:21:02Z)
+   ```
+
+## Applying new configurations
+
+1. Adding a new MachineConfig to a layered MachineConfigPool will cause
+   a new image build to be performed.
+
+2. After the new image is built, rolling out the new image will follow
+   the same process as rolling out new MachineConfigs does. The
+   difference is that the node will be rebased onto the new OS image
+   which contains the new MachineConfigs instead of having the
+   configs written directly to disk.
+
+# Interacting with the MOB/on-cluster builds
+
+## Viewing the Machine OS Builder logs
+
+One can run the following command to view the Machine OS Builder pod logs, if
+the MOB is currently running:
+
+```console
+$ oc logs -f "$(oc get pods -o name -l 'k8s-app=machine-os-builder' -n openshift-machine-config-operator)" -n openshift-machine-config-operator
+```
+
+## Viewing the OS image build logs
+
+Depending on whether one is using the ImageBuildController or the
+PodBuildController, one may view the build logs in one of two ways:
+
+### PodBuildController
+
+```console
+$ oc logs -f -n openshift-machine-config-operator pod/build-<rendered MachineConfig name>
+```
+
+It may be necessary to specify which containers’ logs to view. For example, add
+`-c image-build` to view the image build process or `-c wait-for-done` to view
+the digestfile ConfigMap creation process.
+
+### ImageBuildController
+
+```console
+$ oc logs -f -n openshift-machine-config-operator build/build-<rendered MachineConfig name>
+```
+
+## Viewing the rendered Dockerfile
+
+Whenever a build is currently in progress, the rendered Dockerfile may
+be viewed by running the following command:
+
+```console
+$ oc get configmap/dockerfile-<rendered MachineConfig name> -n openshift-machine-config-operator
+```
+
+However, if the build is successful, this ConfigMap will be deleted upon
+completion of the build process since it is no longer needed.
+
+## Viewing the rendered MachineConfig built into the final OS image
+
+Whenever a build is currently in progress, the rendered MachineConfig
+that is stored within the ConfigMap may be viewed by running:
+
+```console
+$ oc get configmap/mc-<rendered MachineConfig name> -n openshift-machine-config-operator -o json | jq -r '.data["machineconfig.json.gz"]' | base64 -d | gunzip | jq | head
```

Comment:
`oc get configmap/mc-<rendered MachineConfig name> -n openshift-machine-config-operator -o json | jq -r '.data["machineconfig.json.gz"]' | base64 -d | gunzip | jq | head`


I'm sorry, I pasted the "|head"  part of the command that I used to cut the output so that it would be easier to read. We should remove it, or anyone copying the command will not get the full configuration.

Sorry.

Reply:
No worries, I'll fix it!